src/picongpu/include/plugins/particleCalorimeter/ParticleCalorimeter.hpp

```diff
@@ -0,0 +1,530 @@
+#include "particles/policies/ExchangeParticles.hpp"
+#include "math/Vector.hpp"
+#include "algorithms/math.hpp"
+#include <boost/shared_ptr.hpp>
+
+/* libSplash data output */
+#include <splash/splash.h>
+#include <boost/filesystem.hpp>
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <stdlib.h>
+
+namespace picongpu
+{
+using namespace PMacc;
+
+namespace po = boost::program_options;
+
+
+/** Virtual particle calorimeter plugin.
+ *
+ * (virtually) propagates and collects particles to infinite distance.
+ *
+ */
+template<class ParticlesType>
+class ParticleCalorimeter : public ISimulationPlugin
+{
+private:
+    std::string name;
+    std::string prefix;
+    std::string foldername;
+    uint32_t notifyPeriod;
+    MappingDesc* cellDescription;
+    std::ofstream outFile;
+    const std::string leftParticlesDatasetName;
+
+    uint32_t numBinsYaw;
+    uint32_t numBinsPitch;
+    uint32_t numBinsEnergy;
+    float_X minEnergy;
+    float_X maxEnergy;
+    bool logScale;
+    float_X openingYaw_deg;
+    float_X openingPitch_deg;
+    float_X maxYaw_deg;
+    float_X maxPitch_deg;
+
+    float_64 posYaw_deg;
+    float_64 posPitch_deg;
+
+    /* Rotated calorimeter frame */
+    float3_X calorimeterFrameVecX;
+    float3_X calorimeterFrameVecY;
+    float3_X calorimeterFrameVecZ;
+
+    typedef PMacc::container::DeviceBuffer<float_X, DIM3> DBufCalorimeter;
+    typedef PMacc::container::HostBuffer<float_X, DIM3> HBufCalorimeter;
+
+    /* device calorimeter buffer for a single gpu */
+    DBufCalorimeter* dBufCalorimeter;
+    /* device calorimeter buffer for all particles which have left the simulation volume  */
+    DBufCalorimeter* dBufLeftParsCalorimeter;
+    /* host calorimeter buffer for a single mpi rank */
+    HBufCalorimeter* hBufCalorimeter;
+    /* host calorimeter buffer for summation of all mpi ranks */
+    HBufCalorimeter* hBufTotalCalorimeter;
+
+public:
+    typedef CalorimeterFunctor<typename DBufCalorimeter::Cursor> MyCalorimeterFunctor;
+private:
+    typedef boost::shared_ptr<MyCalorimeterFunctor> MyCalorimeterFunctorPtr;
+    MyCalorimeterFunctorPtr calorimeterFunctor;
+
+    typedef boost::shared_ptr<PMacc::algorithm::mpi::Reduce<simDim> > AllGPU_reduce;
+    AllGPU_reduce allGPU_reduce;
+
+    void restart(uint32_t restartStep, const std::string restartDirectory)
+    {
+        if(this->notifyPeriod == 0)
+            return;
+
+        HBufCalorimeter hBufLeftParsCalorimeter(this->dBufLeftParsCalorimeter->size());
+
+        int rank;
+        MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+        if(rank == 0)
+        {
+            splash::SerialDataCollector hdf5DataFile(1);
+            splash::DataCollector::FileCreationAttr fAttr;
+
+            splash::DataCollector::initFileCreationAttr(fAttr);
+            fAttr.fileAccType = splash::DataCollector::FAT_READ;
+
+            std::stringstream filename;
+            filename << restartDirectory << "/" << prefix << restartStep;
```

Comment:
your rebase removed my comment... (this is wasting time, please be more careful)


suggested change:
```C++
filename << restartDirectory << "/" << prefix <<< "_" << restartStep;
```